### PR TITLE
Update to Qt 6.8

### PR DESCRIPTION
**Pull request by nirvn**
In preparation for Qt 6.8.1 which will be what we ship QField 3.5 with.

### CODE REVIEW

File: vcpkg/ports/qtmultimedia/ffmpeg-compile-def.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/src/plugins/multimedia/ffmpeg/CMakeLists.txt b/src/plugins/multimedia/ffmpeg/CMakeLists.txt
+index 77c459a..af5229e 100644
+--- a/src/plugins/multimedia/ffmpeg/CMakeLists.txt
++++ b/src/plugins/multimedia/ffmpeg/CMakeLists.txt
+@@ -273,7 +273,7 @@ if(BUILD_SHARED_LIBS)
+ else()
+     foreach(ffmpeg_lib IN LISTS ffmpeg_libs)
+         qt_internal_add_target_include_dirs(QFFmpegMediaPlugin ${ffmpeg_lib})
+-        target_include_directories(QFFmpegMediaPlugin PRIVATE
++        target_compile_definitions(QFFmpegMediaPlugin PRIVATE
```
Comment (m-kuhn) [inline vcpkg/ports/qtmultimedia/ffmpeg-compile-def.patch:10]: see also https://github.com/microsoft/vcpkg/pull/42080/files#diff-8fdc8f7c211644a960aaf73dcd681d427729564f43c5665c7b20408b63699e22

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 87a691411cf85762c2d0c016ad5a7a2d5d2edc4e: [**arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5880-308e5f71bae7e03c12b89ee410f420f53358bab0-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5880-308e5f71bae7e03c12b89ee410f420f53358bab0-android-arm-neon.apk)
- [x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5880-308e5f71bae7e03c12b89ee410f420f53358bab0-android-x64.apk)
- [x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5880-308e5f71bae7e03c12b89ee410f420f53358bab0-android-x86.apk)
- [**all access arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5880-308e5f71bae7e03c12b89ee410f420f53358bab0-android-arm64.apk)
- [all access arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5880-308e5f71bae7e03c12b89ee410f420f53358bab0-android-arm-neon.apk)
- [all access x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5880-308e5f71bae7e03c12b89ee410f420f53358bab0-android-x64.apk)
- [all access x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5880-308e5f71bae7e03c12b89ee410f420f53358bab0-android-x86.apk)
</details>

File: vcpkg/ports/qtmultimedia/ffmpeg-compile-def.patch (hunk 1 above)
Comment (nirvn) [reply vcpkg/ports/qtmultimedia/ffmpeg-compile-def.patch:10]: @m-kuhn , windows fail to compile, linker complaining about missing ffmpeg functions. Is this linked to that?

Comment (m-kuhn) [reply vcpkg/ports/qtmultimedia/ffmpeg-compile-def.patch:10]: not sure about it if you say windows.
I was expecting problems on android and hence that they would also need to be fixed in the linked PR. And ideally we have an upstreamable patch.

File: cmake/Package.cmake [hunk 2]
```diff
@@ -2,7 +2,7 @@ if(QT_MOC_EXECUTABLE)
   get_filename_component(_qt_bin_dir ${QT_MOC_EXECUTABLE} DIRECTORY)
 endif()
 find_program(LINUXDEPLOY_EXECUTABLE linuxdeploy linuxdeploy-x86_64.AppImage)
-find_program(MACDEPLOYQT_EXECUTABLE macdeployqt HINTS "${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/tools/Qt6/bin" NO_DEFAULT_PATH)
+find_program(MACDEPLOYQT_EXECUTABLE macdeployqt HINTS "${VCPKG_INSTALLED_DIR}/arm64-osx/tools/Qt6/bin" NO_DEFAULT_PATH)
```
Comment (m-kuhn) [inline cmake/Package.cmake:5]: 👀

Comment (nirvn) [reply cmake/Package.cmake:5]: @m-kuhn , not sure if it's caused by updating vcpkg or Qt, but on the x64-osx, we're building both x64 and arm64, the latter is treat as host (like iOS).

**Comment by nirvn** [comment]
Alright, all working except Windows, with the following link errors:

```
[260/272] Linking CXX executable output\bin\qfield.exe
FAILED: output/bin/qfield.exe 
C:\Windows\system32\cmd.exe /C "cd . && D:\a\_temp\542323655\cmake-3.29.0-windows-x86_64\bin\cmake.exe -E vs_link_exe --intdir=src\app\CMakeFiles\qfield.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100261~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100261~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\Llvm\x64\bin\lld-link.exe /nologo @CMakeFiles\qfield.rsp  /out:output\bin\qfield.exe /implib:src\app\qfield.lib /pdb:output\bin\qfield.pdb /version:0.0 /machine:x64 /INCREMENTAL:NO /subsystem:console  /DEBUG /OPT:REF /OPT:ICF /machine:x64 && C:\Windows\system32\cmd.exe /C "cd /D C:\builddir\src\app && "C:\Program Files\PowerShell\7\pwsh.exe" -noprofile -executionpolicy Bypass -file C:/builddir/_deps/vcpkg-src/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary C:/builddir/output/bin/qfield.exe -installedDir C:/builddir/vcpkg_installed/x64-windows-static/bin -OutVariable out""
LINK: command "C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\Llvm\x64\bin\lld-link.exe /nologo @CMakeFiles\qfield.rsp /out:output\bin\qfield.exe /implib:src\app\qfield.lib /pdb:output\bin\qfield.pdb /version:0.0 /machine:x64 /INCREMENTAL:NO /subsystem:console /DEBUG /OPT:REF /OPT:ICF /machine:x64 /MANIFEST:EMBED,ID=1" failed (exit code 1) with the following output:
lld-link: error: undefined symbol: av_log_get_level
>>> referenced by C:\src\qtmultimedia\src\here-src-6-7497dd0ae1.clean\src\plugins\multimedia\ffmpeg\qffmpegmediaintegration.cpp:103
>>>               ffmpegmediaplugin.lib(qffmpegmediaintegration.cpp.obj):(void __cdecl qffmpegLogCallback(void *, int, char const *, char *))
```

File: cmake/qgis-cmake-wrapper.cmake [hunk 3]
```diff
@@ -142,7 +142,22 @@ if(TRUE) # Should possibly have a "static only" check
...
+  message(STATUS ${AVCODEC_LIBRARY})
+  message(STATUS ${AVDEVICE_LIBRARY})
+  message(STATUS ${AVFORMAT_LIBRARY})
+  message(STATUS ${SWRESAMPLE_LIBRARY})
+  message(STATUS ${SWSCALE_LIBRARY})
+  target_link_libraries(QGIS::Core INTERFACE ${AVCODEC_LIBRARY})
+  target_link_libraries(QGIS::Core INTERFACE ${AVDEVICE_LIBRARY})
+  target_link_libraries(QGIS::Core INTERFACE ${AVFORMAT_LIBRARY})
+  target_link_libraries(QGIS::Core INTERFACE ${SWRESAMPLE_LIBRARY})
+  target_link_libraries(QGIS::Core INTERFACE ${SWSCALE_LIBRARY})
```
Comment (m-kuhn) [inline cmake/qgis-cmake-wrapper.cmake:160]: If ever that makes it to prod, this should be trimmed down to 

```
find_package(FFMPEG COMPONENTS avcodec avdevice avformat swresample swscale REQUIRED)
  target_link_libraries(QGIS::Core INTERFACE
    ${AVCODEC_LIBRARY}
    ${AVDEVICE_LIBRARY}
    ${AVFORMAT_LIBRARY}
    ${SWRESAMPLE_LIBRARY}
    ${SWSCALE_LIBRARY}
  )
```

**Comment by nirvn** [comment]
@m-kuhn , this is ready for final review / merge.

File: cmake/Package.cmake (hunk 2 above)
Comment (m-kuhn) [reply cmake/Package.cmake:5]: I think there should be something like this in `QT_HOST_PATH`. I'd keep it at least as a comment that we took the shortcut here.

File: cmake/Platform.cmake [hunk 4]
```diff
@@ -22,7 +22,7 @@ endif()
 if(ANDROID_ABI)
   set(ANDROID_BUILD_TOOLS_VERSION "34.0.0" CACHE STRING "Android build-tools version")
   set(ANDROID_TARGET_PLATFORM 34 CACHE INT "Target Android platform SDK version")
-  set(ANDROID_PLATFORM 23 CACHE INT "Minimum Android platform SDK version")
-  set(ANDROID_PLATFORM_INT 23 CACHE INT "Minimum Android platform SDK version") # Used in build.gradle.in
+  set(ANDROID_PLATFORM 26 CACHE INT "Minimum Android platform SDK version")
+  set(ANDROID_PLATFORM_INT 26 CACHE INT "Minimum Android platform SDK version") # Used in build.gradle.in
```
Comment (m-kuhn) [inline cmake/Platform.cmake:26]: Can we use `ANDROID_PLATFORM` in build.gradle.in?

Comment (nirvn) [reply cmake/Platform.cmake:26]: It wasn't possible a few years ago as far value was being modified and added some string suffix

File: cmake/Package.cmake (hunk 2 above)
Comment (nirvn) [reply cmake/Package.cmake:5]: +1 to comment

**Review by m-kuhn — APPROVED** [review]
Let's merge to get testing
Very good work here

**Pull request merged**